File: omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/pod/response/ActivationResponseBase.java
package info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.response;

abstract class ActivationResponseBase extends ResponseBase {
    final ResponseType.ActivationResponseType activationResponseType;

    ActivationResponseBase(ResponseType.ActivationResponseType activationResponseType, byte[] encoded) {
        super(ResponseType.ACTIVATION_RESPONSE, encoded);
        this.activationResponseType = activationResponseType;
    }

    public ResponseType.ActivationResponseType getActivationResponseType() {
        return activationResponseType;
    }
}

File: pump/omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/pod/response/ActivationResponseBase.kt
package info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.response

import info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.pod.response.ResponseType.ActivationResponseType

abstract class ActivationResponseBase(
    val activationResponseType: ActivationResponseType,
    encoded: ByteArray
) : ResponseBase(ResponseType.ACTIVATION_RESPONSE, encoded)
